django_test_migrations/db/checks/statement_timeout.py

```diff
@@ -0,0 +1,104 @@
+import datetime
+from typing import List
+
+from django.core.checks import CheckMessage, Info, Warning
+from django.db import connections
+from typing_extensions import Final
+
+from django_test_migrations.db.backends import exceptions, registry
+from django_test_migrations.db.backends.base.configuration import (
+    BaseDatabaseConfiguration,
+)
+from django_test_migrations.logic.datetime import timedelta_to_miliseconds
+from django_test_migrations.types import AnyConnection
+
+#: We use this value as a unique identifier of databases related check.
+CHECK_NAME: Final = 'django_test_migrations.checks.database_configuration'
+STATEMENT_TIMEOUT_MINUTES_UPPER_LIMIT: Final = 30
+
+
+def check_statement_timeout_setting(*args, **kwargs) -> List[CheckMessage]:
+    """Check if statements' timeout settings is properly configured."""
+    messages: List[CheckMessage] = []
+    for connection in connections.all():
+        _check_statement_timeout_setting(connection, messages)
+    return messages
+
+
+def _check_statement_timeout_setting(
+    connection: AnyConnection,
+    messages: List[CheckMessage],
+) -> None:
+    try:
+        database_configuration = registry.get_database_configuration(
+            connection,
+        )
+    except exceptions.DatabaseConfigurationNotFound:
+        return
+
+    try:
+        setting_value = int(database_configuration.get_setting_value(
+            database_configuration.statement_timeout,
+        ))
+    except exceptions.DatabaseConfigurationSettingNotFound:
+        return
+
+    _ensure_statement_timeout_is_set(
+        database_configuration,
+        setting_value,
+        messages,
+    )
+    _ensure_statement_timeout_not_too_high(
+        database_configuration,
+        setting_value,
+        messages,
+    )
+
+
+def _ensure_statement_timeout_is_set(
+    database_configuration: BaseDatabaseConfiguration,
+    setting_value: int,
+    messages: List[CheckMessage],
+) -> None:
+    if not setting_value:
+        messages.append(
+            Warning(
```

Comment:
Let's discuss what should be a `Warning` and what should be an `Error`?
What do you think?

Reply:
We just want to give some suggestions to users that they need to take closer look at or adjust some DB settings, so probably we should use more `Warning` s than `Error` s, but this case is quite good for `Error` (I will change it), because not setting `statement_timeout` value at all is quite risky (as described in blog post linked to original issue).

But the second check message used in this file is quite tricky, because there is no universal upper limit for `statement_timeout` expression it's highly related to concrete cases/apps, that's the reason I have used `Info`, but as you mentioned in one of comments it's impossible to automate in CI/CD pipelines, so we probably need to increment message level by one, so:

+ `django_test_migrations.checks.autonames.W001` - `Warning` becomes `E001`  - `Error` - autonames are really bad practice so we can treat it as errors
+ `django_test_migrations.checks.database_configuration.I001` - `Info` becomes `W001` - `Warning` - upper limit is hard to guess, so we can only give some hint to the developers
+ `django_test_migrations.checks.database_configuration.W001` - `Warning` becomes `E001` - `Error` - it's risky to not set `statement_timeout`

Generally:

+ `Error` - when bad configuration can have serious consequences or setting some value is required
+ `Warning` - when something is usually required, but hard to guess without some domain/internal knowledge, so developers should analyze it themselves 
+ `Info` - some notes, suggestions to user that aren't critical 

What's your opinion?